ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveUtils.java

```diff
@@ -518,11 +518,11 @@ public static Path getDumpPath(Path root, String dbName, String tableName) {
   }
 
   public static String getTableSnapshotRef(String refName) {
-    Matcher ref = SNAPSHOT_REF.matcher(refName);
-    if (ref.matches()) {
-      return ref.group(1);
+    if (org.apache.commons.lang3.StringUtils.isEmpty(refName)) {
```

Comment:
you can just do `SNAPSHOT_REF.matcher(String.valueOf(refName))` and keep the rest as it is

Reply:
Good!